front-end/src/utils/deviceInformation.ts

```diff
@@ -0,0 +1,30 @@
+import { useEffect, useState } from "react"
+
+/** Device breakpoints for responsive design for tailwind css */
+const deviceBreakpoints = {
+  sm: 640,
+  md: 768,
+  lg: 1024,
+  xl: 1280,
+  "2xl": 1536,
+}
+
+/**
+ * This function checks if the device is smaller than the preferred breakpoint.
+ *
+ * @returns hasSpecificDeviceSize - if the device is smaller than or equal to the preferred breakpoint
+ */
+export function hasSpecificDeviceSize(breakpoint: keyof typeof deviceBreakpoints): boolean {
```

Comment:
Das ist ein Hook und Hooks müssen immer mit `use` anfangen, also `useDeviceSize()`. Es wundert mich, dass das so derzeit funktioniert.

Ich würde auch eher `sm`, `md` zurückgeben, also statt Rückgabewert `boolean` lieber `keyof typeof deviceBreakpoints`

Reply:
Außerdem in das Verzeichnis verschieben, wo auch die anderen Hooks sind.